src/app/pages/Ontology.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { useState, useEffect } from 'react';
+import { fetchDocument, fetchDocuments, APIDocument, updateOntology } from '../lib/api';
+import { useHistory, Link } from 'react-router-dom';
+import { Console } from 'console';
+
+function Ontology() {
+    const [docs, setDocs] = useState<APIDocument>();
+    const [name, setName] = useState<string>('');
+    const [pack, setPack] = useState<string>('');
+    const [ontology, setOntology] = useState<string>('');
+    const [edit_state, setEditState] = useState<string>('');
+    const history = useHistory();
+
+    useEffect(() => {
+        updateDocs().catch(e => {
+          history.push('/login');
+        });
+      }, [history]);
+    
+      function updateDocs() {
+        
+        let doc_id = window.location.pathname.split("/").pop();
+
+        if (doc_id != undefined){
+            return fetchDocument(doc_id).then(docs => {
+                setOntology(docs.ontology);
+            });
+        }
+        else{
+            return fetchDocuments().then(docs => {
+                setDocs(docs);
+            });
+        }
+    }
+
+
+  function redirectSignup(e: any) {
+    history.push("/signup") 
+  }
+
+  function handleEditOntology() {
+      let doc_id = window.location.pathname.split("/").pop();
+      if (doc_id != undefined){
+       updateOntology(doc_id , ontology).then(edit_state => {
+        updateDocs();
+        setEditState(edit_state);
+        if(edit_state.status == 1){
+            console.log('Save OK');
+        }
+        else{
+            console.log('Save Failed'); 
+        }
+        
+       });   
+      }
+  }
+
+  console.log(ontology);
```

Comment:
I think you can remove console calls after your own debug.

Reply:
removed